File: comms/bulk-messaging/sender-pools/list-sender-pools/list-sender-pools.rb
require "twiliocomms"

client = TwilioComms::Client.new(
  account_id: "TWILIO_ACCOUNT_SID",
  auth_token: "TWILIO_AUTH_TOKEN"
)

client.sender_pools.list(
  start_date: "2024-01-15T09:30:00Z",
  end_date: "2024-01-15T09:30:00Z",
  operation_id: "comms_operation_01h9krwprkeee8fzqspvwy6nq8",
  page_token: "pageToken",
  page_size: 50
)

File: comms/bulk-messaging/sender-pools/list-sender-pools/list-sender-pools.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\SenderPools\Requests\SenderPoolsListRequest;
use DateTime;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->senderPools->list(
    new SenderPoolsListRequest([
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'operationId' => 'comms_operation_01h9krwprkeee8fzqspvwy6nq8',
        'pageToken' => 'pageToken',
        'pageSize' => 50,
    ]),
);
